components/style/style_adjuster.rs

```diff
@@ -84,10 +82,9 @@ impl<'a, 'b: 'a> StyleAdjuster<'a, 'b> {
         let blockified_display =
             display.equivalent_block_display(flags.contains(CascadeFlags::IS_ROOT_ELEMENT));
         if display != blockified_display {
-            self.style.mutate_box().set_adjusted_display(
-                blockified_display,
-                is_item_or_root,
-            );
+            if let Some(box_) = self.style.get_box_if_mutated() {
+                box_.set_original_display(blockified_display);
```

Comment:
How is this ok in any way? This is completely changing the output.

Before this change: `original_display: display, display: blockified_display`.
After this change: `original_display: blockified_display, display: display`.

That looks pretty wrong.

Reply:
Also, it's not ok to just do it if the `display` struct has mutated, this needs to mutate it unconditionally.

Reply:
> Before this change: `original_display: display, display: blockified_display`.
> After this change: `original_display: blockified_display, display: display`.

I think we still need `is_item_or_root` inside `set_original_display` of servo part.

With the current patch, it will set both `display` and `original_display` directly.
With passing `blockified_display`, I think the `after this change` will be `original_display: blockified_display, display: blockified_display`.

Let me try to fix it.